src/server/node_services/nodes_monitor.js

```diff
@@ -3494,7 +3495,12 @@ class NodesMonitor extends EventEmitter {
                 'issues_report', item.node.issues_report,
                 'block_report', block_report);
             // disconnect from the node to force reconnect
-            this._disconnect_node(item);
+            // only disconnect if enough time passed since last disconnect to avoid amplification of errors in R\W flows
+            const DISCONNECT_GRACE_PERIOD = 2 * 60 * 1000; // 2 minutes grace before another disconnect
```

Comment:
Maybe add this to config.js?

Reply:
It’s something very specific to this flow. I usually prefer to put it close to the usage in these cases